pkg/acsengine/engine.go

```diff
@@ -1008,13 +1008,27 @@ func getDCOSMasterCustomNodeLabels() string {
 
 func getDCOSAgentCustomNodeLabels(profile *api.AgentPoolProfile) string {
 	var buf bytes.Buffer
+    var attr_string string
 	buf.WriteString("")
+    if len(profile.OSType) > 0  {
```

Comment:
./engine.go:1028: invalid operation: profile.Ports != "" (mismatched types []int and string)

See, osType is a string, but not a string